### PR TITLE
add documentation for migration guide

### PR DESCRIPTION
**Pull request by jay-deshmukh**
- FIx https://tyktech.atlassian.net/browse/TT-1621

### CODE REVIEW

**Review by marksou — APPROVED** [review]

**Review by vaske — APPROVED** [review]

File: tyk-docs/content/graphql/migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+- To improve performance now a single Data Source can be used to link to multiple fields instead of having an independent data source for every field hence `graphql.type_field_configurations` is now obsolete and new data sources can be defined under `graphql.engine.data_sources` (see example below).
+
+- Data Source kind are `REST` or `GraphQL` regardless of api being internal or not.
+
+- Incase of internal apis `graphql.engine.data_sources[n].internal` property is set to true.
+
+- Each dataSources needs to be defined with a unique name `graphql.engine.data_sources[n].name`.
+
+- Each field connected to the data source is expected to be configured for mapping under `graphql.engine.field_configs`
```
Comment (devsergiy) [inline tyk-docs/content/graphql/migration-guide.md]: @jay-deshmukh should we say here that the field `should be` configured nevertheless it is requires mapping or not?

File: tyk-docs/content/graphql/migration-guide.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+date: 2020-06-03
+menu:
+  main:
+    parent: "GraphQL"
+weight: 1
+aliases:
+    - /graphql/migration/
+---
+
+As of 3.2 graphql schema under tyk api definition (i.e `api_definition.graphql`) changed significantly, hence graphql api definitions created in previous beta versions are not supported via UI and need to go through following changes manually.
```
Comment (pvormste) [inline tyk-docs/content/graphql/migration-guide.md:12]: Maybe we should mention explicitly that old api definitions will continue to work for the gateway

File: tyk-docs/content/graphql/migration-guide.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+- To improve performance now a single Data Source can be used to link to multiple fields instead of having an independent data source for every field hence `graphql.type_field_configurations` is now obsolete and new data sources can be defined under `graphql.engine.data_sources` (see example below).
+
+- Data Source kind are `REST` or `GraphQL` regardless of api being internal or not.
+
+- Incase of internal apis `graphql.engine.data_sources[n].internal` property is set to true.
```
Comment (pvormste) [inline tyk-docs/content/graphql/migration-guide.md]: do we need to explain, that internal APIs are the ones that are accessed via `tyk://` scheme?

**Comment by emma-m-campbell** [comment]
@jay-deshmukh From a user perspective, we're looking at directing people from the dashboard to this docs page so that they can make the necessary changes. Could be just my lack of technical knowledge but I'm not sure that this describes enough what the user should do. It describes what changes have happened and why but not so much specifically what I need to change in the raw definition. Maybe it could use wording like "you need to remove this and replace it with this, or edit this to this" and then once 'x' is done then it will be unbroken and you can use the dashboard again? Could we get a Tyk developer from outside the manage squad to read this and see if they understand what needs to be done to unbreak the changes?

**Comment by jay-deshmukh** [comment]
@ConsM @ifrim can you take a look at this with your fresh eyes please, and confirm if it's clear how to redefine schema. 
Thanks

**Comment by ConsM** [comment]
From my little knowledge, guidance around data sources look good.
Maybe we need a similar approach for old and new graphql definitions ?

File: tyk-docs/content/graphql/migration-guide.md [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+**Note**  
+Old Api definitions will continue to work for the gateway
+
+
+- To improve performance now a single Data Source can be used to link to multiple fields instead of having an independent data source for every field hence `graphql.type_field_configurations` is now obsolete and new data sources can be defined under `graphql.engine.data_sources` (see example below).
+
+- Data Source kind are `REST` or `GraphQL` regardless of api being internal or not.
+
+- Incase of internal apis that are accessed via `tyk://`scheme, `graphql.engine.data_sources[n].internal` property is set to true.
```
Comment (ifrim) [inline tyk-docs/content/graphql/migration-guide.md]: I think a space is missing here.
`In case` instead of `Incase`.

**Comment by ifrim** [comment]
@jay-deshmukh After reading this I think that I would be confident enough that I would know what needs to be changed.

**Comment by jay-deshmukh** [comment]
thanks @ifrim and @ConsM 🙏

**Comment by marksou** [comment]
Merged

**Pull request merged**